src/Chirp.Infrastructure/Repository/ReactionRepository.cs

```diff
@@ -0,0 +1,65 @@
+using Chirp.Core.Entities;
+using Chirp.Core.Repository;
+using Microsoft.EntityFrameworkCore;
+
+namespace Chirp.Infrastructure.Repository;
+
+public class ReactionRepository: BaseRepository, IReactionRepository
+{
+    public ReactionRepository(ChirpDbContext dbContext) : base(dbContext)
+    {
+    }
+    
+    public async Task AddReaction(ReactionType reaction, Guid cheepId, Guid authorId)
+    {
+        Cheep? cheep = await db.Cheeps.FindAsync(cheepId);
+        Author? author = await db.Users.FindAsync(authorId);
+        if (cheep != null)
+        {
+            Reaction entity = new Reaction()
+            {
+                ChirpId = cheepId,
+                Cheep = cheep,
+                AuthorId = authorId,
+                Author = author,
+                ReactionType = reaction
+            };
+            db.Reactions.Add(entity);
+            await db.SaveChangesAsync();
+        }
+        else
+        {
+            //
+            throw new Exception("Cheep with id " + cheepId + " not found");
+        }
+    }
+    public async Task RemoveReaction(ReactionType reaction, Guid cheepId, Guid authorId)
+    {
+            Reaction? entity = await db.Reactions.FindAsync(cheepId, authorId);
+            if (entity != null) db.Reactions.Remove(entity);
+            await db.SaveChangesAsync();
```

Comment:
Saving changes is only necessary when making changes. Move 'await db.SaveChangesAsync();' into the id-statement clause